autogen/agentchat/contrib/retrieve_user_proxy_agent.py

```diff
@@ -372,31 +365,27 @@ def retrieve_docs(self, problem: str, n_results: int = 20, search_string: str =
         """
         if not self._collection or not self._get_or_create:
             print("Trying to create collection.")
-            self._client = create_vector_db_from_dir(
-                dir_path=self._docs_path,
+            retriever_class = get_retriever(self._retriever_type)
+            self.retriever = retriever_class(
+                name=self._collection_name,
+                embedding_model_name=self._embedding_model,
+                embedding_function=self._embedding_function,
                 max_tokens=self._chunk_token_size,
-                client=self._client,
-                collection_name=self._collection_name,
                 chunk_mode=self._chunk_mode,
                 must_break_at_empty_line=self._must_break_at_empty_line,
-                embedding_model=self._embedding_model,
-                get_or_create=self._get_or_create,
-                embedding_function=self._embedding_function,
                 custom_text_split_function=self.custom_text_split_function,
+                use_existing=not self._get_or_create,
+                client=self._client,
                 custom_text_types=self._custom_text_types,
                 recursive=self._recursive,
             )
             self._collection = True
-            self._get_or_create = True
-
-        results = query_vector_db(
-            query_texts=[problem],
-            n_results=n_results,
-            search_string=search_string,
-            client=self._client,
-            collection_name=self._collection_name,
-            embedding_model=self._embedding_model,
-            embedding_function=self._embedding_function,
+            self._get_or_create = False
+        self.retriever.ingest_data(self._docs_path)
```

Comment:
Do we need to `ingest_data` at every single call of `retrieve_docs`?

Reply:
Ohh not really. I think that what's the code was doing before, no? I'll just check if the index already exists and only create it if it doesn't?

Reply:
In the old code, `create_vector_db_from_dir` happens only under the condition `if not self._collection or not self._get_or_create:`, `ingest_data` is a part of `create_vector_db_from_dir`, looks like better to be under the same condition. WDYT?

Reply:
Btw, how do you `check if the index already exists`? I think it would be interesting if we could check if there are new contents in the `docs_path` and automatically add the new contents into the same collection. This could be addressed in a new PR.

Reply:
@thinkall I've made the change, can you please take a look again?